l10n_it_riba_sale_commission/models/account_move.py

```diff
@@ -1,9 +1,30 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
+from datetime import timedelta
+
 from odoo import fields, models
 
 
 class AccountMove(models.Model):
     _inherit = "account.move"
 
     no_commission = fields.Boolean(string="Without commissions")
+
+
+class AccountInvoiceLineAgent(models.Model):
+    _inherit = "account.invoice.line.agent"
+
+    def _skip_future_payments(self, date_payment_to):
+        if self.invoice_id.is_riba_payment:
+            return any(
+                any(
+                    date_payment_to
+                    < line.date_maturity
+                    + timedelta(distinta_line.riba_line_id.config_id.safety_days)
+                    for distinta_line in line.distinta_line_ids
+                )
+                for line in self.invoice_id.line_ids
+                if line.date_maturity
+            )
```

Comment:
probabilmente qui potresti prendere il massimo valore di: 

`line.date_maturity + timedelta(distinta_line.riba_line_id.config_id.safety_days)` 

in modo da skippare il pagamento fino a che la data maturity più distante è stata raggiunta.  

Potrebbe essere un'idea?